### PR TITLE
[MINOR] docs: Add benchmark results

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?
Add the benchmark data.

### Why are the changes needed?
Attract more users. If we have benchmark data, users don't need to test by themselves.  It will lower the barrier to use.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Just doc

### CODE REVIEW

**Comment by zuston** [comment]
Great improvement. 

How about adding `Netty` ?

**Comment by jerqi** [comment]
> Great improvement.
> 
> How about adding `Netty` 
We should update the bencmark after finishing Netty feature.

File: docs/benchmark.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+  ## Environment
+  Software: Uniffle 0.2.0  Hadoop 2.8.5 Spark 2.4.6
+  Hardware: Machine 176 cores, 265G memory, 4T * 12 HDD, network bandwidth 10GB/s
+  Hadoop Yarn Cluster: 1 * ResourceManager + 6 * NodeManager, every machine 4T * 10 HDD
+  Uniffle Cluster: 1 * Coordinator + 6 * Shuffle Server, every machine 4T * 10 HDD
```
Comment (kaijchen) [inline docs/benchmark.md:21]: please fix the format here.

File: docs/benchmark.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  |query91|17|21|
+  |query92|12|12|
+  |query93|86|86|
+  |query94|40|42|
+  |query95|94|94|
+  |query96|10|10|
+  |query97|29|34|
+  |query98|17|21|
+  |query99|13|12|
+  |total|5821|6494|
```
Comment (kaijchen) [inline docs/benchmark.md:148]: Add an newline after the table

File: docs/benchmark.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  ![Write Time](asset/rss_benchmark2.png)
+  Read Time:
+  ![Read Time](asset/rss_benchmark1.png)
+  #### vanilla Spark performance
+  Overall Time:
+  ![Overall Time](asset/vanilla_benchmark1.png)
+  Write Time:
+  ![Write Time](asset/vanilla_benchmark2.png)
+  Read Time:
+  ![Read Time](asset/vanilla_benchmark3.png)
```
Comment (kaijchen) [inline docs/benchmark.md:166]: These screenshots looks blurry, is it possible to put raw data?

File: docs/benchmark.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  spark.rss.storage.type MEMORY_LOCALFILE
+  ````
+  Shuffle Server's configuration
+  ````
+  rss.storage.type MEMORY_LOCALFILE
+  rss.server.buffer.capacity 50g
+  ````
+  
+  ## TPC-DS
+  We generate 1TB data, we use the code of the [repo](https://github.com/databricks/spark-sql-perf)
```
Comment (kaijchen) [inline docs/benchmark.md:40]:
```suggestion
  We used [spark-sql-perf](https://github.com/databricks/spark-sql-perf) to generate 1TB data.
```

**Review by kaijchen — COMMENTED** [review]
Thanks @jerqi for the benchmark result. Maybe we should add some charts along with the table to show the result intuitively.

**Comment by jerqi** [comment]
> Thanks @jerqi for the benchmark result. Maybe we should add some charts along with the table to show the result intuitively.
 I tried. But there are too many queries. Charts won't give more effective information.

File: docs/benchmark.md (hunk 1 above)
Comment (kaijchen) [reply docs/benchmark.md:21]: > What format do you prefer?

You are missing line breaks here, see preview.

File: docs/benchmark.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+  |query93|86|86|
+  |query94|40|42|
+  |query95|94|94|
+  |query96|10|10|
+  |query97|29|34|
+  |query98|17|21|
+  |query99|13|12|
+  |total|5821|6494|
+  
+  Uniffle is a little 9% slower than vanilla Spark. Because the amount of shuffle is tiny.
```
Comment (jiafuzha) [inline docs/benchmark.md:150]: The perf is lower  than vanilla Spark. Is it better to remove TPC-DS bench for now? We can add it back when we have better perf with large shuffle data.

**Comment by awdavidson** [comment]
It would be good to see the uniffle coordinator and shuffle-server configuration used when performing these benchmark e.g. heap, buffer sizes etc

File: docs/benchmark.md (hunk 1 above)
Comment (jerqi) [reply docs/benchmark.md]: done.

**Comment by jerqi** [comment]
@zuston @kaijchen @jiafuzha I think we should approve this pr first and merge it. It will be better than nothing. When `Netty` feature is finished, we can update benchmark results.

**Comment by connorlwilkes** [comment]
1TB is quite a small dataset, has anyone benchmarked more than this and if so what did the results/config look like?

**Comment by jerqi** [comment]
> 1TB is quite a small dataset, has anyone benchmarked more than this and if so what did the results/config look like?

What dataset do you hope to add?

**Review by zuston — APPROVED** [review]
Let's merge this firstly

**Comment by jerqi** [comment]
Thanks all, merged.

**Pull request merged**